.github/workflows/rust.yml

```diff
@@ -0,0 +1,55 @@
+name: Rust
+
+on:
+  push:
+    branches: [ main ]
+  schedule:
+    - cron: 0 0 1 * *
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_INCREMENTAL: 1
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macos-latest
+        include:
+          - os: ubuntu-latest
+            name: Linux
+            lint: 1
+          - os: windows-latest
+            name: Windows
+          - os: macos-latest
+            name: MacOS
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: cargo build (${{ matrix.name }})
```

Comment:
It might be easier to use matrix.os and remove the name entirely from the matrix